doc/developer/platform/ux.md

```diff
@@ -1,20 +1,902 @@
+variable:
+
+```sql
+SET cluster = <name>;
+```
+
+Users can configure a default cluster for their sessions. How this default is
+configured is TODO.
+
+Access to a cluster is gated by SQL permission grants. Users cannot execute
+queries on a cluster on which they do not have `USAGE` privileges.
+
+### Database
+
+A database is a simple container of schemas that can be subject to access
+control. It confers no additional semantics.
+
+The only supported privilege is `CREATE`, which allows users to create
+schemas in the database.
+
+TODO: should databases be associated with a single [timeline](#timeline)?
+
+### Schema
+
+A schema is a simple container of database objects that can be subject to access
+control. It confers no additional semantics.
+
+The only supported privilege is `CREATE`, which allows users to create objects
+in the schema.
+
+Databases and schemas are simple namespaces to sallow for downstream
+organizations and tools to develop opinions about how to structure a Materialize
+Platform project.
+
+For example, an organization might achieve separate production and staging
+environments by creating a `production` database and a `staging` database, then
+using [roles](#role) to limit access to `production` to a trusted set of
+engineers and service accounts.
+
+### Source
+
+The user experience of a source is largely unchanged.
+
+The existing `CREATE MATERIALIZED SOURCE` statement is extended to support
+specifying the cluster in which to create the index:
+
+```
+CREATE MATERIALIZED SOURCE <name> FROM ... [IN CLUSTER <cluster>]
+```
+
+If no cluster is explicitly specified, the active cluster is used. If there is
+no active cluster, the statement is rejected.
+
+Users cannot read data from a source on which they do not have the `SELECT`
+privilege. Note that if the source references a [secret](#secret), privileges on
+that secret are not considered when determining whether a user has access to the
+source.
+
+### Sink
+
+Like indexes, sinks cause computation to occur, and therefore must be bound to a
+cluster.
+
+There are no relevant privileges on sinks.
+
+We do not yet know how to provide sinks in Materialize Platform. We are,
+however, aware of at least two important user experience constraints:
+
+  * The default sink experience must be "exactly once." If a cluster restarts,
+    downstream sinks must not reexport data that was previously exported.
+
+  * Sinks must be prepared for the behavior of the COMPUTE layer to change after
+    a version upgrade. If we fix a bug in COMPUTE that changes the data computed
+    by an index, sinks must detect and communicate this behavior change to
+    users. Options include marking the sink as errored and asking users to
+    recreate the sink, or emitting the diff between the old computation and
+    new computation and then proceeding with export of new values.
+
+### Table
+
+The user experience of a table is unchanged.
```

Comment:
How do tables related to COMPUTE vs. STORAGE? Does this mean that contents of tables will be tied to indexes?

Reply:
Currently, internally, TABLE is a lot like a read-once materialized source. There is a TABLE and there is also always an INDEX, but they are two separate things. Platform makes this clearer, where STORAGE gets to record the TABLE, rather than having the INDEX do it.

Reply:
I'm not seeing the UX change! Won't you be able to type `SELECT * FROM storage_collection`, where `storage_collection` is a source _or_ a table, and we'll go fetch that collection for you, even though you've not built an index on it?

Reply:
Oh, hm, I guess they persist their data. That's a pretty huge change. Oops! Fixed.

Reply:
The ones I was thinking of are minor, like "we will build a dataflow for you that isn't built now" and "we can return indexed reads for you". Sorry, should have given examples!